syft/pkg/cataloger/javascript/parse_package_lock.go

```diff
@@ -47,38 +47,51 @@ func parsePackageLock(path string, reader io.Reader) ([]*pkg.Package, []artifact
 	var packages []*pkg.Package
 	dec := json.NewDecoder(reader)
 
+	var lock PackageLock
 	for {
-		var lock PackageLock
 		if err := dec.Decode(&lock); err == io.EOF {
 			break
 		} else if err != nil {
 			return nil, nil, fmt.Errorf("failed to parse package-lock.json file: %w", err)
 		}
-		licenseMap := make(map[string]string)
-		for _, pkgMeta := range lock.Packages {
-			var sb strings.Builder
-			sb.WriteString(pkgMeta.Resolved)
-			sb.WriteString(pkgMeta.Integrity)
-			licenseMap[sb.String()] = pkgMeta.License
-		}
+	}
 
+	if lock.LockfileVersion == 1 {
 		for name, pkgMeta := range lock.Dependencies {
-			var sb strings.Builder
-			sb.WriteString(pkgMeta.Resolved)
-			sb.WriteString(pkgMeta.Integrity)
-			var licenses []string
-			if license, exists := licenseMap[sb.String()]; exists {
-				licenses = append(licenses, license)
-			}
 			packages = append(packages, &pkg.Package{
 				Name:     name,
 				Version:  pkgMeta.Version,
 				Language: pkg.JavaScript,
 				Type:     pkg.NpmPkg,
-				Licenses: licenses,
 			})
 		}
 	}
 
+	if lock.LockfileVersion == 2 || lock.LockfileVersion == 3 {
+		for name, pkgMeta := range lock.Packages {
+			if name == "" {
```

Comment:
I wonder if we should include this? We'd need to read the `name` field, it's an indication what this package itself is...

Reply:
Is there a place it should be read to? One of the return types?

Reply:
I think just added to the `Dependency` struct above. And then `if name == "" && pkgMeta.Name != "" { name = pkgMeta.Name }` or so

Reply:
Just following up on this. I'm not sure I follow the solution; how does Syft make use of the name of the root package? Do I return it somewhere?

Reply:
Let's say we `git clone github.com/facebook/react` and then we `syft dir:react`, we wouldn't have the _react_ package anywhere, even though we were directly scanning `react`. I think we should make packages from this "root"

Reply:
_Packages_ aren't dependencies -- they are _the things found when scanning_.

Reply:
Well, sure, but if you say "scan my application" do we typically list the application as a package?

Reply:
(Thats not rhetorical, I have no idea how Syft works currently)

Reply:
If you say: "scan this node binary" we return "we found a node binary at version X", if we can. There is some inconsistence here, to be frank, but I think we should definitely include this stuff if we have the information. For a `go` binary, for example: it may not actually include the root project name, so we only have dependencies to surface.

Reply:
Okay, I've updated to include the root name